crates/syntax/src/ast/make.rs

```diff
@@ -294,6 +294,21 @@ pub fn param_list(pats: impl IntoIterator<Item = ast::Param>) -> ast::ParamList
     ast_from_text(&format!("fn f({}) {{ }}", args))
 }
 
+pub fn generic_param(name: String, ty: Option<String>) -> ast::GenericParam {
```

Comment:
We usually try to use signatures which match the syntax tree closely:

```
I: Foo + Bar

->

TYPE_PARAM@321..333
  NAME@321..322
    IDENT@321..322 "I"
  COLON@322..323 ":"
  WHITESPACE@323..324 " "
  TYPE_BOUND_LIST@324..333
    TYPE_BOUND@324..327
      PATH_TYPE@324..327
        PATH@324..327
          PATH_SEGMENT@324..327
            NAME_REF@324..327
              IDENT@324..327 "Foo"
    WHITESPACE@327..328 " "
    PLUS@328..329 "+"
    WHITESPACE@329..330 " "
    TYPE_BOUND@330..333
      PATH_TYPE@330..333
        PATH@330..333
          PATH_SEGMENT@330..333
            NAME_REF@330..333
              IDENT@330..333 "Bar"
```

So, a more appropriate signature would be `generic_param(name: ast::Name, type_bound_list: Option<ast::TypeBoundList>) `